src/dhcpmon/src/dhcp_device.c

```diff
@@ -498,36 +573,42 @@ void dhcp_device_shutdown(dhcp_device_context_t *context)
 }
 
 /**
- * @code dhcp_device_get_status(context);
+ * @code dhcp_device_get_status(check_type, context);
  *
  * @brief collects DHCP relay status info for a given interface. If context is null, it will report aggregate
  *        status
  */
-dhcp_mon_status_t dhcp_device_get_status(dhcp_device_context_t *context)
+dhcp_mon_status_t dhcp_device_get_status(dhcp_mon_check_t check_type, dhcp_device_context_t *context)
 {
-    dhcp_mon_status_t rv = 0;
+    dhcp_mon_status_t rv = DHCP_MON_STATUS_HEALTHY;
 
     if (context != NULL) {
-        rv = dhcp_device_validate(context->counters, context->counters_snapshot);
-        memcpy(context->counters_snapshot, context->counters, sizeof(context->counters_snapshot));
-    } else {
-        rv = dhcp_device_validate(glob_counters, glob_counters_snapshot);
-        memcpy(glob_counters_snapshot, glob_counters, sizeof(glob_counters_snapshot));
+        rv = dhcp_device_check_health(check_type, context->counters, context->counters_snapshot);
     }
 
     return rv;
 }
 
 /**
- * @code dhcp_device_print_status();
+ * @code dhcp_device_update_snapshot(context);
+ *
+ * @brief Update device/interface counters snapshot
+ */
+void dhcp_device_update_snapshot(dhcp_device_context_t *context)
+{
+    if (context != NULL) {
+        memcpy(context->counters_snapshot, context->counters, sizeof(context->counters_snapshot));
```

Comment:
I thought context->counters_snapshot = context->counters will do the same?

Reply:
These are arrays of structs. Recent patch changed them into arrays.